macros/sql/pivot.sql

```diff
@@ -0,0 +1,62 @@
+{#
+Pivot values from rows to columns.
+
+Example:
+
+    Input:
+
+    | size | color |
+    |------+-------|
+    | S    | red   |
+    | S    | blue  |
+    | S    | red   |
+    | M    | red   |
+
+    select
+      size,
+      {{ pivot('size', ['red', 'blue']) }}
+    from <table>
+    group by size
+
+    Output:
+
+    | size | red | blue |
+    |------+-----+------|
+    | S    | 2   | 1    |
+    | M    | 1   | 0    |
+
+Arguments:
+    column: Column name, required
+    values: List of row values to turn into columns, required
+    alias: Whether to create column aliases, default is True
+    op: SQL aggregation function, default is sum
+    cmp: SQL value comparison, default is =
+    prefix: Column alias prefix, default is blank
+    postfix: Column alias postfix, default is blank
+    then_value: Value to use if comparison succeeds, default is 1
+    else_value: Value to use if comparison fails, default is 0
+#}
+
+{% macro pivot(column,
+               values,
+               alias=True,
+               op='sum',
+               cmp='=',
+               prefix='',
+               postfix='',
+               then_value=1,
+               else_value=0) %}
+  {% for v in values %}
+    {{ op }}(
+      case
+      when {{ column }} {{ cmp }} '{{ v }}'
+        then {{ then_value }}
+      else {{ else_value }}
+      end
+    )
+    {% if alias %}
+      as {{ prefix + v + postfix }}
```

Comment:
If you use the jinja concatenation operator (`~`) instead of the addition operator (`+`), then this will work when `v` is numeric too. Otherwise, Jinja throws:

```
2018-01-10 19:25:18,553: Compilation Error in macro pivot (pivot.sql)
  must be str, not int
```

Reply:
It's also a good idea to quote `{{ prefix ~ v ~ suffix }}` here. Otherwise, if there's no prefix and `v` is not a string, this will break.

To quote in a cross-platform way, you can do:

```
      as {{ adapter.quote(prefix ~ v ~ suffix) }}
```

So this will work for Redshift, Snowflake, and BigQuery